src/plugins/d3/renderer/components/Legend.tsx

```diff
@@ -17,76 +18,80 @@ type Props = {
 
 export const Legend = (props: Props) => {
     const {width, offsetWidth, height, offsetHeight, chartSeries, onItemClick} = props;
+    const ref = React.useRef<SVGGElement>(null);
 
-    return (
-        <g
-            width={width}
-            height={height}
-            ref={(node) => {
-                if (!node) {
-                    return;
-                }
+    React.useEffect(() => {
+        if (!ref.current) {
+            return;
+        }
 
-                const size = 10;
-                const textWidths: number[] = [0];
-                const svgElement = select(node);
-                svgElement.selectAll('*').remove();
-                const legendItemTemplate = svgElement
-                    .selectAll('legend-history')
-                    .data(chartSeries)
-                    .enter()
-                    .append('g')
-                    .attr('class', b('item'))
-                    .on('click', function (e, d) {
-                        onItemClick({name: d.name, metaKey: e.metaKey});
-                    });
-                svgElement
-                    .selectAll('*')
-                    .data(chartSeries)
-                    .append('text')
-                    .text(function (d) {
-                        return d.name;
-                    })
-                    .each(function () {
-                        textWidths.push(this.getComputedTextLength());
-                    })
-                    .remove();
+        const legendItems = (
+            chartSeries.length === 1 && !isAxisRelatedSeries(chartSeries[0])
+                ? chartSeries[0].data
+                : chartSeries
+        ) as {color: string; name: string; visible?: boolean}[];
```

Comment:
Maybe it 's better to transfer this to the PreparedLegend calculation ?  And to pass to the component only the necessary data in the required format.